examples/08_join_aggregation.py

```diff
@@ -1,302 +1,548 @@
+# We define some plotting functions to display our results.
+import seaborn as sns
+from matplotlib import pyplot as plt
+from sklearn.calibration import CalibrationDisplay
+
+
+def plot_gain_tradeoff(results):
+    """Scatter plot of the score gain (y) vs the fit time (x) for each model."""
+
+    rows = []
+    for estimator_name, result in results.items():
+        result["estimator_name"] = estimator_name
+        rows.append(result)
+    df = pd.DataFrame(rows)
+
+    names = df["estimator_name"].values
+    palette = dict(zip(names, sns.color_palette("colorblind", n_colors=len(names))))
+
+    fig, ax = plt.subplots(figsize=(5, 4), dpi=100)
+    sns.scatterplot(
+        df,
+        x="time_to_fit",
+        y="gain_score",
+        hue="estimator_name",
+        style="estimator_name",
+        ax=ax,
+        palette=palette,
+        s=200,
+    )
+    ax.grid()
+
+    ticks = df["time_to_fit"].round(3).tolist()
+    labels = [f"{tick}s" for tick in ticks]
+    ax.set_xticks(ticks, labels)
+
+    ticks = df["gain_score"].round().tolist()
+    ticks.insert(1, 650_000)
+    labels = [f"{tick:,} €" for tick in ticks]
+
+    ax.set_yticks(ticks, labels)
+    ax.set_ylabel("Gain score")
+    ax.set_xlabel("Time to fit")
+    ax.set_title("Gain score vs Time to fit")
+    plt.tight_layout()
+
+
+def plot_calibration_curve(results):
+    """Plot a calibration curve and the log-loss."""
+
+    estimator_names = list(results)
+    palette = dict(
+        zip(
+            estimator_names,
+            sns.color_palette("colorblind", n_colors=len(estimator_names)),
+        )
+    )
+    fig, ax = plt.subplots(figsize=(6, 4), dpi=100)
+    for name, result in results.items():
+        log_loss = str(round(result["log_loss"], 4))
+        label = f"{name}, {'log_loss: ' + log_loss}"
+        CalibrationDisplay.from_predictions(
+            y_true=result["y_test"],
+            y_prob=result["y_proba"],
+            strategy="quantile",
+            label=label,
+            ax=ax,
+            color=palette[name],
+            n_bins=15,
+        )
+    ax.set_xlim([-0.001, 0.13])
+    ax.set_ylim([-0.001, 0.13])
+    ax.set_title("Calibration curve")
+
+
+# %%
+# We see below that the low effort classifier significantly improves our gains compared
+# to the dummy baseline. The former is of course slower to train than the latter.
+
+plot_gain_tradeoff(results)
+
+
+# %%
+# We also evaluate the calibration of both models. As very few classes are
+# positive, we can expect all probabilities to be close to 0. We have to
+# zoom on it, and use the "quantile" strategy of |CalibrationDisplay| in order to create
+# bins containing an equal number of samples.
+
+plot_calibration_curve(results)
+
+
+# %%
+# Agg-Joiner based estimator
+# --------------------------
 #
-# Note that you need the name of the pipeline elements to assign them
-# hyper-parameters search.
+# We first need to split the dataframe between a dataframe representing baskets and a
```

Comment:
maybe more of a transition here. it seemed like we were done with our analysis and now we jump with "we first need" and it's not clear why.

but to make the example shorter I think we could split the tables in the fetcher and start the example here

Reply:
although it is interesting to see that the "correct" approach using the aggjoiner is faster _and_ more effective